src/core/thread/key_manager.cpp

```diff
@@ -633,7 +633,23 @@ void KeyManager::SetNetworkKeyRef(otNetworkKeyRef aKeyRef)
 exit:
     return;
 }
-
 #endif // OPENTHREAD_CONFIG_PLATFORM_KEY_REFERENCES_ENABLE
 
+void KeyManager::DeleteSecurityKeys(bool aClearPersistentKeys)
+{
+    mMleKey.Clear();
+    mKek.Clear();
+    Get<Mac::SubMac>().DeleteMacKeys();
+    Get<Mac::Mac>().DeleteMode2Key();
```

Comment:
I am Ok either way. II chose this way, because it will be cleaner and it really doesn't affect the device behaviour if RAM keys are cleared. But if you prefer we isolate this to PSA, I am ok with it.

Reply:
Thanks. If we want to clear RAM, may be better to clear everything. We can do `memset(instance, 0, sizeof(otInstance))` in `Finalize` (we can consider this in a future PR).